src/DocumentFormat.OpenXml.Framework/EnumerableExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DocumentFormat.OpenXml;
+
+internal static class EnumerableExtensions
+{
+    /// <summary>
+    /// Similar to <see cref="Enumerable.FirstOrDefault{TSource}(IEnumerable{TSource})"/> but will also verify that at most there is one.
+    /// </summary>
+    public static T? FirstOrDefaultAndMaxOne<T>(this IEnumerable<T> enumerable, Func<Exception>? exceptionFactory = null)
```

Comment:
Should it name as `SingleOrDefault xxx` ?

Reply:
Good catch! I had forgotten about that method - it's the same thing :)

Reply:
@twsouthwick did you want to change this? I'll approve but wait to merge.

Reply:
haha it automerged. Replaced it in https://github.com/dotnet/Open-XML-SDK/pull/1899